### PR TITLE
Add new OIDC attributes to DCR

### PR DESCRIPTION
**Pull request by SachiniSiriwardene**
PR - https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2173

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/OAuthDCRMTestCase.java [hunk 1]
```diff
@@ -314,4 +315,76 @@ private String getAuthzHeader() {
...
+        obj.put(OAuthDCRMConstants.JWKS_URI, "https://localhost:9443/oauth2/jwks");
+
+        StringEntity entity = new StringEntity(obj.toJSONString());
+        request.setEntity(entity);
+
+        HttpResponse response = client.execute(request);
+        assertEquals(response.getStatusLine().getStatusCode(), 201, "Service Provider " +
+                "has not been created successfully");
+
+        JSONObject createResponsePayload  = getPayload(response);
```
Comment (janakamarasena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/OAuthDCRMTestCase.java:368]: Lets assert that all the attributes that should be returned has returned as well.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/OAuthDCRMTestCase.java [hunk 2]
```diff
@@ -314,4 +315,76 @@ private String getAuthzHeader() {
...
+
+        BufferedReader rd = new BufferedReader(new InputStreamReader(response.getEntity().getContent()));
+        Object responseObj = JSONValue.parse(rd);
+        EntityUtils.consume(response.getEntity());
+        return (JSONObject) responseObj;
+    }
+
+    @Test(alwaysRun = true, groups = "wso2.is", priority = 9, description = "Create a service provider with " +
+            "additional OIDC properties")
+    public void testCreateServiceProviderRequestWithAdditionalParameters() throws IOException {
```
Comment (janakamarasena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/OAuthDCRMTestCase.java:328]: Shall we add a test for updates as well?

Comment (SachiniSiriwardene) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/OAuthDCRMTestCase.java:329]: added

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/OAuthDCRMTestCase.java (hunk 1 above)
Comment (SachiniSiriwardene) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/OAuthDCRMTestCase.java:369]: fixed

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/util/OAuthDCRMConstants.java [hunk 3]
```diff
@@ -46,6 +46,23 @@ public class OAuthDCRMConstants {
...
+    public static final String REQUEST_OBJECT_SIGNATURE_ALGORITHM = "request_object_signing_alg";
+    public static final String TLS_SUBJECT_DN = "tls_client_auth_subject_dn";
+    public static final String IS_PUSH_AUTH = "require_pushed_authorization_requests";
+    public static final String IS_SIGNED_REQUEST_OBJECT = "require_signed_request_object";
+    public static final String IS_CERTIFICATE_BOUND_ACCESS_TOKEN = "tls_client_certificate_bound_access_tokens";
+    public static final String SUBJECT_TYPE = "subject_type";
+    public static final String REQUEST_OBJECT_ENCRYPTION_ALGORITHM = "request_object_encryption_alg";
+    public static final String REQUEST_OBJECT_ENCRYPTION_METHOD = "request_object_encryption_enc";
+
+
```
Comment (janakamarasena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/util/OAuthDCRMConstants.java:66]: Remove unnecessary empty lines

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/util/OAuthDCRMConstants.java [hunk 4]
```diff
@@ -46,6 +46,23 @@ public class OAuthDCRMConstants {
     public static final String GRANT_TYPE_AUTHORIZATION_CODE = "authorization_code";
     public static final String GRANT_TYPE_PASSWORD = "password";
 
+    public static final String TOKEN_AUTH_METHOD =  "token_endpoint_auth_method";
+    public static final String TOKEN_AUTH_SIGNATURE_ALGORITHM = "token_endpoint_auth_signing_alg";
+    public static final String SECTOR_IDENTIFIER_URI = "sector_identifier_uri";
+    public static final String ID_TOKEN_SIGNATURE_ALGORITHM = "id_token_signed_response_alg";
+    public static final String ID_TOKEN_ENCRYPTION_ALGORITHM = "id_token_encrypted_response_alg";
+    public static final String ID_TOKEN_ENCRYPTION_METHOD = "id_token_encrypted_response_enc";
+    public static final String AUTH_RESPONSE_SIGNATURE_ALGORITHM = "authorization_signed_response_alg";
```
Comment (janakamarasena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/util/OAuthDCRMConstants.java:55]: Didn't we discuss to remove this?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/util/OAuthDCRMConstants.java (hunk 3 above)
Comment (SachiniSiriwardene) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/util/OAuthDCRMConstants.java:66]: fixed

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/util/OAuthDCRMConstants.java (hunk 4 above)
Comment (SachiniSiriwardene) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/util/OAuthDCRMConstants.java]: removed from code... removed from constants as well

**Review by janakamarasena — APPROVED** [review]

**Comment by SachiniSiriwardene** [comment]
Closing since new tests are addedt through pr - https://github.com/wso2/product-is/pull/17047

**Pull request closed without merging**